src/libcharon/sa/ikev2/tasks/ike_rekey.c

```diff
@@ -139,6 +139,7 @@ static void establish_new(private_ike_rekey_t *this)
 		}
 #ifdef USE_KERNEL_SUPER
 		charon->kernel->write_terminator_tunnel_ike_spis(charon->kernel, this->new_sa);
+		charon->kernel->write_terminator_tunnel_ike_sa_auth_uptime(charon->kernel, this->new_sa);
```

Comment:
1. This is a rekey task, need to call `write_terminator_tunnel_ike_sa_uptime()`
2. Also, are you sure that on IKE rekey the lifetime needs to be reset? The requirements are vague about it